woocommerce/woocommerce
# Fix display of "empty cart" notices when using a shortcode cart and the checkout block

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

C&C Blocks introduced some new notice markup via some updated templates. These templates use a different HTML structure than the notices in core. The core notices get a special classname added in `wc_empty_cart_message` called `cart-empty`, but this is not added to the new Block notices because of the updated markup. For reference:

```
$notice  = str_replace( 'class="woocommerce-info"', 'class="cart-empty woocommerce-info"', $notice );
```

Cart.js uses the `cart-empty` to target elements from ajax responses. This is handled within `update_wc_div`. Because the `cart-empty` class does not exist when using blocks, this update fails and so after interacting with the shortcode cart, the page is emptied. 

The fix in this PR is to wrap everything returned by `wc_empty_cart_message` with a new `wc-empty-cart-message` classname, which is then used by cart.js so the update works in both contexts.

Closes #38719



### How to test the changes in this Pull Request:

This needs to be tested twice to ensure there are no regressions.

**Test 1:**
For the pages assigned as cart/checkout under WooCommerce > Settings > Advanced:
- ensure that the Cart Page contains only the [woocommerce_cart] shortcode
- ensure that the Checkout Page contains only the [woocommerce_checkout] shortcode

1. Add an item to the cart
2. View the **cart page**
3. Remove all items from the cart using the X link
4. The page should show the following:

![Screenshot 2023-06-15 at 11 42 01](https://github.com/woocommerce/woocommerce/assets/90977/30183dd2-1bcf-488b-8671-24706bc1f333)

**Test 2**
For the pages assigned as cart/checkout under WooCommerce > Settings > Advanced:
- ensure that the Cart Page contains only the [woocommerce_cart] shortcode
- ensure that the Checkout Page contains only the **Checkout Block**

1. Add an item to the cart
2. View the cart page
3. Remove all items from the cart using the X link
4. The page should show the following:

![Screenshot 2023-06-15 at 11 41 31](https://github.com/woocommerce/woocommerce/assets/90977/03730e3c-49ba-4c42-a841-288098db221c)

Notice the styling is updated between tests.
